packages/xstate-store/src/store.ts

```diff
@@ -346,3 +590,16 @@ export function createStoreTransition<
 function uniqueId() {
   return Math.random().toString(36).slice(6);
 }
+
+type MaybeZod<T> = T extends { _output: infer U } ? U : never;
+
+export type EmittedFromSchemas<TSchemas extends StoreSchemas> =
+  TSchemas extends {
+    emitted: infer TEmitted;
+  }
+    ? {
+        [K in keyof TEmitted]: Compute<
+          { type: K } & MaybeZod<TEmitted[K]> & EventObject
```

Comment:
`EventObject` doesn't bring any value here:
```suggestion
          { type: K } & MaybeZod<TEmitted[K]>
```

Reply:
Typechecking fails without this